shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/repository/SolrRepositoryImpl.java

```diff
@@ -19,147 +26,125 @@
-					criteria.and(Criteria.where(DATASET_TYPE).contains(datasetType.substring(1, datasetType.length() -1)));
-				} else if (datasetType.startsWith(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_TYPE).endsWith(datasetType.substring(1, datasetType.length())));
-				} else if (datasetType.endsWith(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_TYPE).startsWith(datasetType.substring(0, datasetType.length() - 1)));
-				}
-			}
-		}
-		if (facet.getDatasetNature() != null && !facet.getDatasetNature().isEmpty()) {
-			for (String datasetNature: facet.getDatasetNature()) {
-				if(!datasetNature.contains(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_NATURE).is(facet.getDatasetNature()));
-				} else if (datasetNature.startsWith(WILDCARD) && datasetNature.endsWith(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_NATURE).contains(datasetNature.substring(1, datasetNature.length() -1)));
-				} else if (datasetNature.startsWith(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_NATURE).endsWith(datasetNature.substring(1, datasetNature.length())));
-				} else if (datasetNature.endsWith(WILDCARD)) {
-					criteria.and(Criteria.where(DATASET_NATURE).startsWith(datasetNature.substring(0, datasetNature.length() - 1)));
-				}
-			}
+		
+		if (facet.getSearchText() != null && !facet.getSearchText().trim().isEmpty()) {
+			if (facet.isExpertMode()) {
+				addExpertClause(criteria, facet.getSearchText());	
+			} else {
+				addSearchInAllClause(criteria, facet.getSearchText());			
+			}			
 		}
 
 		criteria = combineCriteria(criteria);
 
 		SimpleFacetQuery query = ((FacetQuery) new SimpleFacetQuery(criteria)
 				.setPageRequest(pageable))
-				.setFacetOptions(new FacetOptions().addFacetOnField("studyName_str")
-						.addFacetOnField("subjectName_str")
-						.addFacetOnField("datasetName_str")
-						.addFacetOnField("examinationComment_str")
-						.addFacetOnField(DATASET_TYPE)
-						.addFacetOnField(DATASET_NATURE)
+				.setFacetOptions(new FacetOptions()
+						.addFacetOnField(STUDY_NAME_FACET)
+						.addFacetOnField(SUBJECT_NAME_FACET)
+						.addFacetOnField(DATASET_NAME_FACET)
+						.addFacetOnField(EXAMINATION_COMMENT_FACET)
+						.addFacetOnField(DATASET_TYPE_FACET)
+						.addFacetOnField(DATASET_NATURE_FACET)
 						.setFacetLimit(-1));
 
-		SolrResultPage<ShanoirSolrDocument> result = (SolrResultPage<ShanoirSolrDocument>) solrTemplate.queryForPage(query, ShanoirSolrDocument.class);
-		FacetPage<ShanoirSolrDocument> result2 = solrTemplate.queryForFacetPage(query, ShanoirSolrDocument.class);
-
-
-		return (SolrResultPage<ShanoirSolrDocument>) result2;
+		try {
+			FacetPage<ShanoirSolrDocument> result = solrTemplate.queryForFacetPage(query, ShanoirSolrDocument.class);			
+			return (SolrResultPage<ShanoirSolrDocument>) result;
+		} catch (UncategorizedSolrException e) {
+			ErrorModel error = new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "solr query failed");
```

Comment:
The exception is never logged here, it could be interesting for us to keep track of it in the logs.

Reply:
It's supposed to be logged when the RestServiceException is catch

Reply:
Yes but you never use the initial exception, in you want it to be logged, you should do 
throw new RestServiceException(error, e); I think